core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -97,6 +97,7 @@ class KafkaApis(val requestChannel: RequestChannel,
         case requestId => throw new KafkaException("Unknown api code " + requestId)
       }
     } catch {
+      case fee: FatalExitError => throw fee
```

Comment:
Yeah it makes sense and seems much cleaner than the previous approach of using an anonymous thread.